File: Data Structures & Algorithms/maximum-subarray/submission-3.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        res, total = float('-inf'), 0
        r = 0
        while r < len(nums):
            total += nums[r]
            res = max(res, total)
            if total < 0:
                total = 0
            r += 1
        
        return res

        

File: Data Structures & Algorithms/jump-game/submission-3.java
class Solution {
    public boolean canJump(int[] nums) {
        boolean[] canRead = new boolean[nums.length];
        canRead[0] = true;
        
        for(int i = 0; i < nums.length; i++) {
            if (canRead[i]) {
                for (int j = 0; j <= nums[i]; j++) {
                    if (i+j < nums.length) {
                        canRead[i+j] = true;
                    }   
                }
            }
        }

        return canRead[canRead.length-1];
    }
}
